plugins/modules/cloud/docker/docker_compose.py

```diff
@@ -767,7 +767,7 @@ def cmd_up(self):
                         ))
                     result['actions'].append(result_action)
 
-        if not self.check_mode and result['changed']:
+        if not self.check_mode and result['changed'] and not self.stopped:
```

Comment:
What happens if the service is modified? Does it still get updated? Did you test this with `pull: yes` and `restarted: yes`?

Reply:
Updated? No because if `stopped: yes` is set the server will not be started and therefor got not updated. It will of course get the new config running after the stopped flag was removed and the service will be started again.

Reply:
`restarted` and `stopped` should IMO never work together, that does not really makes sense, right?

Reply:
The change does not affect the logic for `pull: yes` in any case, I don't really understand the question... The pull logic as handled before the changed line will be reached https://github.com/ansible-collections/community.general/blob/c69a94aab4616fb37f6ab056c9100b787ed66195/plugins/modules/cloud/docker/docker_compose.py#L726

Reply:
I agree, that `restarted` and `stopped` should not work together. My question is whether or not that specific case was tested. What happens if they are both `yes`?

As for pull... yes, that's exactly why I asked. If `pull` is set to `yes`, then `result['changed']` can be `true`, and the container will be started. Your code as written should work to stop the container from being started, but I wanted to make sure you tested it.

Reply:
Ok... Even if I think the stopped/restart discussion is unrelated to this issue I've added a condition now to prevent the usage of these parameters at the same time.

Tested cases:

- `pull: yes` will download the new image but will not start the container if `stopped: yes` is set (expected)
-  service modifications will not trigger a restart if `stopped: yes` is set (expected)
- `restarted: yes` and `stopped: yes` will no longer work at the same time

Reply:
Thanks! Good work on this!

Reply:
Thanks! Good work on this!